Frontend/src/views/Home/Home.jsx

```diff
@@ -0,0 +1,82 @@
+import { Fragment } from 'react';
+import JoinChat from '../JoinChat/JoinChat';
+import './Home.css';
+import socket from '../../services/socket';
+import { useState } from 'react';
+import { useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+function Home({ setShouldSocketConnect }) {
+  const navigate = useNavigate();
+  console.log('home component');
+  const [connectedUsers, setConnectedUsers] = useState(new Map());
+  useEffect(() => {
+    // Add event listener for the "users" event
+    socket.emit('users');
+
+    socket.on('users', (users) => {
+      console.log('users event', users);
+      setConnectedUsers(new Map(users));
+    });
+
+    socket.on('newUsers', (user) => {
+      addUser(user);
+    });
+
+    socket.on('userDisconnected', (user) => {
+      console.log('userdisconnected event occur', user);
+      removeUser(user.id);
+    });
+
+    // Clean up the event listener on component unmount
+    return () => {
+      socket.off('users');
+    };
+  }, []);
+
+  const handleLeave = () => {
+    sessionStorage.removeItem('userToken');
+    socket.disconnect();
+    setShouldSocketConnect(false);
+    navigate('/');
+  };
+  const genrateUniqueId = () => {
```

Comment:
use UUID for unique ID instead of math random number there should be a probability of repeating random number